crypto/s2n_composite_cipher_aes_sha.c

```diff
@@ -167,17 +167,21 @@ static int s2n_composite_cipher_aes_sha_encrypt(struct s2n_session_key *key, str
     POSIX_ENSURE_EQ(out->size, in->size);
 
     POSIX_GUARD_OSSL(EVP_EncryptInit_ex(key->evp_cipher_ctx, NULL, NULL, NULL, iv->data), S2N_ERR_KEY_INIT);
-    POSIX_GUARD_OSSL(EVP_Cipher(key->evp_cipher_ctx, out->data, in->data, in->size), S2N_ERR_ENCRYPT);
+
+    int len = out->size;
```

Comment:
More of an observation

len is an out only parameter.
https://github.com/openssl/openssl/blob/openssl-3.0.5/crypto/evp/evp_enc.c#L607
https://github.com/openssl/openssl/blob/openssl-3.0.5/crypto/evp/evp_enc.c#L756

Currently in the code there are four cases where it is not assigned a value:
https://github.com/aws/s2n-tls/blob/fb1e57c104eea441f0a928bfa15eca6979810b00/crypto/s2n_aead_cipher_chacha20_poly1305.c#L64-L66
https://github.com/aws/s2n-tls/blob/fb1e57c104eea441f0a928bfa15eca6979810b00/crypto/s2n_aead_cipher_chacha20_poly1305.c#L102-L104
https://github.com/aws/s2n-tls/blob/f9d4618bb37e5de38260d6e4e6916e3c29cbf56c/crypto/s2n_aead_cipher_aes_gcm.c#L225-L227
https://github.com/aws/s2n-tls/blob/f9d4618bb37e5de38260d6e4e6916e3c29cbf56c/crypto/s2n_aead_cipher_aes_gcm.c#L263-L265
Five cases where it is assigned a value:
https://github.com/aws/s2n-tls/blob/fe8df74ddafb712d3141056107db7228b19b6ef7/crypto/s2n_cbc_cipher_3des.c#L37-L38
https://github.com/aws/s2n-tls/blob/fe8df74ddafb712d3141056107db7228b19b6ef7/crypto/s2n_cbc_cipher_3des.c#L50-L51
https://github.com/aws/s2n-tls/blob/fe8df74ddafb712d3141056107db7228b19b6ef7/crypto/s2n_cbc_cipher_aes.c#L42-L43
https://github.com/aws/s2n-tls/blob/fe8df74ddafb712d3141056107db7228b19b6ef7/crypto/s2n_cbc_cipher_aes.c#L42-L43
https://github.com/aws/s2n-tls/blob/247bf3151194c3326e1c9f63cb1b1c06ab30f1ea/crypto/s2n_drbg.c#L51-L52

Reply:
The `outl` param is overwritten by the OSSL calls so setting an initial value is not necessary. It possibly causes some confusion as well.

Reply:
cleaned up in https://github.com/aws/s2n-tls/pull/3466/commits/b4abd23ea44c900b28b34319cba49066c10ebd20